core/forkchoice.go

```diff
@@ -86,9 +86,16 @@ func (f *ForkChoice) ReorgNeeded(current *types.Header, extern *types.Header) (b
 		localTD  = f.chain.GetTd(current.Hash(), current.Number.Uint64())
 		externTd = f.chain.GetTd(extern.Hash(), extern.Number.Uint64())
 	)
-	if localTD == nil || externTd == nil {
+	if localTD == nil {
 		return false, errors.New("missing td")
 	}
+	if externTd == nil {
+		ptd := f.chain.GetTd(extern.ParentHash, extern.Number.Uint64()-1)
```

Comment:
is it possible to hit this case? or have you observed it before?

Reply:
got it, it can be occured after this PR, as the block has not written to db yet.

Reply:
> is it possible to hit this case? or have you observed it before?

when execute updateHighestVerifiedHeader
td still not write down